FTBFS - Boost 1.64.0 -  serialization::make_array

With boost 1.64.0, make_array was removed(?) from boost::serialization.
numeric::ublas was not updated for this change, thus freeorion will fail to compile (only first error shown):
```console
CMakeFiles/freeorioncommon.dir/build.make:110: recipe for target 'CMakeFiles/freeorioncommon.dir/combat/CombatLogManager.cpp.o' failed
In file included from /opt/projects/boost/1_64_0/include/boost/numeric/ublas/vector.hpp:21:0,
                 from /opt/projects/boost/1_64_0/include/boost/numeric/ublas/matrix.hpp:18,
                 from /opt/projects/boost/1_64_0/include/boost/numeric/ublas/symmetric.hpp:16,
                 from /opt/projects/freeorion/combat/../universe/Universe.h:11,
                 from /opt/projects/freeorion/combat/CombatSystem.h:4,
                 from /opt/projects/freeorion/combat/CombatLogManager.h:4,
                 from /opt/projects/freeorion/combat/CombatLogManager.cpp:1:
/opt/projects/boost/1_64_0/include/boost/numeric/ublas/storage.hpp: In member function ‘void boost::numeric::ublas::unbounded_array<T, ALLOC>::serialize(Archive&, unsigned int)’:
/opt/projects/boost/1_64_0/include/boost/numeric/ublas/storage.hpp:299:18: error: ‘make_array’ is not a member of ‘boost::serialization’
             ar & serialization::make_array(data_, s);
                  ^~~~~~~~~~~~~
```

Boost ticket [#12978](https://svn.boost.org/trac/boost/ticket/12978)
There is a (still pending) upstream PR: [boostorg/ublas #46](https://github.com/boostorg/ublas/pull/46), inclusion of which would allow build to complete.

I have not yet found the commit in boost::serialization that made this change, so posting as issue in case someone comes across it before.

A potential workaround:
```diff
diff --git a/universe/Universe.h b/universe/Universe.h
index d98d8d6..9ada5c3 100644
--- a/universe/Universe.h
+++ b/universe/Universe.h
@@ -8,6 +8,9 @@
 
 #include <boost/signals2/signal.hpp>
 #include <boost/unordered_map.hpp>
+#if BOOST_VERSION >= 106400
+    #include <boost/serialization/array_wrapper.hpp>
+#endif
 #include <boost/numeric/ublas/symmetric.hpp>
 #include <boost/serialization/access.hpp>
 #include <boost/thread/shared_mutex.hpp>
```

--- a/universe/Universe.h
+++ b/universe/Universe.h
@@ -8,6 +8,12 @@
 
 #include <boost/signals2/signal.hpp>
 #include <boost/unordered_map.hpp>
+// Fix for issue #1513 (boost ticket #12978)
+// Starting with v1.64, boost::numeric::ublas was not updated for changes to
+// boost::serialization, which moved array_wrapper to separate header
+#if BOOST_VERSION >= 106400
+    #include <boost/serialization/array_wrapper.hpp>
+#endif
 #include <boost/numeric/ublas/symmetric.hpp>
 #include <boost/serialization/access.hpp>
 #include <boost/thread/shared_mutex.hpp>

